### ISSUE TITLE
Hide che-theia menu items using theia extension

### ISSUE THREAD

**Issue by priyamariarao** [eclipse-che/che#18466]
This is a follow up question to 
https://github.com/eclipse/che/issues/18439

I want to hide the che-theia menu items using an extension. 
I am able to build the Theia extension and use the contributes menu to hide the menu items and this works fine on the browser-app. I package this extension as a vsix and successfully able to add it to the plugin registry. the plugin also shows as "installed" but it doesn't hide the menu items as expected. I dont see any error logs. Can you please point me to what I can do to debug this issue

**Comment by sunix** [comment]
Have you tried to use the vsix file in plain Theia?
You can have a look to the menu `view > output`.
Could you share a sample of your extension that we could try?

**Comment by azatsarynnyy** [comment]
Hello @priyamariarao 
please, let me know if something wasn't clear in the following comment https://github.com/eclipse/che/issues/18439#issuecomment-733224087 I'll rephrase it.

Actually, you're trying to mix two distinct concepts.
`vsix` packaging is required for Che Theia Plug-ins. You may think of it as VC Code extensions.
But in your case, you're developing Che Theia extension. It's doesn't require any packaging.

**Comment by priyamariarao** [comment]
@azatsarynnyy thank you for the update. I actually need to create an extension that can be loaded when the workspace is loaded. Is it possible to hide the main menu items using the VS code extension. If yes can you guide me how to do it?

**Comment by azatsarynnyy** [comment]
@priyamariarao Plugin API, which is VS Code compatible, doesn't allow to manipulate with the main menu

**Comment by priyamariarao** [comment]
I have a che instance hosted with a theia image as part of the deployment. so if any user launches a workspace they will have a theia IDE by default. I now need to customize the theia menu. After I build the theia extension how can I make it available for all users by default @azatsarynnyy .

**Comment by azatsarynnyy** [comment]
@priyamariarao you need to build your own Che Theia container with your extension included.

One of the possible ways is:
- fork [che-theia repo](https://github.com/eclipse/che-theia)
- add your extension to [che-theia-init-sources.yml](https://github.com/eclipse/che-theia/blob/master/che-theia-init-sources.yml) config file in the same way how it's done for other extensions
- rebuild the image with `./build.sh` script. More details on the possible build options are in [the doc](https://github.com/eclipse/che-theia/tree/master/dockerfiles/theia#how-to-build-theia-image)

**Comment by sunix** [comment]
the steps are:
- customise https://github.com/eclipse/che-theia/blob/master/che-theia-init-sources.yml
- rebuild the che-theia container image with your image
- push it in dockerhub or quay.io
- change the default editor in your che instance: `CHE_WORKSPACE_DEVFILE_DEFAULT__EDITOR` https://www.eclipse.org/che/docs/che-7/installation-guide/advanced-configuration-options-for-the-che-server-component/#LDevfile-defaults

**Comment by priyamariarao** [comment]
Thank you @azatsarynnyy @sunix . I am trying out the suggested options

**Comment by priyamariarao** [comment]
> * the doc

I tried this way. I tried to build the theia image locally before including the extension. I get the below error. please let me know what I am missing

Cloning into '/home/theia-dev/theia-source-code'...
Your branch is up to date with 'origin/master'.
Removing intermediate container 9447020c20da
 ---> 42fb8432e82e
Step 11/38 : ADD src/patches ${HOME}/patches
 ---> dae61d6a06c2
Step 12/38 : RUN if [ -d "${HOME}/patches/${THEIA_VERSION}" ]; then     echo "Applying patches for Theia version ${THEIA_VERSION}";     for file in $(find "${HOME}/patches/${THEIA_VERSION}" -name '*.patch'); do       echo "Patching with ${file}";       cd ${HOME}/theia-source-code && patch -p1 < ${file};     done     fi
 ---> Running in 48bed9e1746e
Removing intermediate container 48bed9e1746e
 ---> c36ac7dcb149
Step 13/38 : ARG CDN_PREFIX=""
 ---> Running in 29b476b49b5f
Removing intermediate container 29b476b49b5f
 ---> 14bdef09ae21
Step 14/38 : ARG MONACO_CDN_PREFIX=""
 ---> Running in 372d36abff96
Removing intermediate container 372d36abff96
 ---> e1e217b23400
Step 15/38 : WORKDIR ${HOME}/theia-source-code
 ---> Running in fafe227d4ad9
Removing intermediate container fafe227d4ad9
 ---> 94f2c15db658
Step 16/38 : COPY asset-che-theia.tar.gz /tmp/asset-che-theia.tar.gz
 ---> 5badfe05a72c
Step 17/38 : RUN mkdir -p ${HOME}/theia-source-code/che-theia/ && tar xzf /tmp/asset-che-theia.tar.gz -C ${HOME}/theia-source-code/che-theia/ && rm /tmp/asset-che-theia.tar.gz
 ---> Running in f0471388b48c
Removing intermediate container f0471388b48c
 ---> 71d9586aa7e8
Step 18/38 : RUN che-theia init -c ${HOME}/theia-source-code/che-theia/che-theia-init-sources.yml --alias https://github.com/eclipse/che-theia=${HOME}/theia-source-code/che-theia
 ---> Running in 14d505709b1a
/home/theia-dev/.yarn-global/node_modules/@eclipse-che/theia-generator/dist
Error: Config file does not exists
    at new CliError (/home/theia-dev/.yarn-global/node_modules/@eclipse-che/theia-generator/dist/cli-error.js:33:28)
    at InitSources.<anonymous> (/home/theia-dev/.yarn-global/node_modules/@eclipse-che/theia-generator/dist/init-sources.js:492:35)
    at step (/home/theia-dev/.yarn-global/node_modules/@eclipse-che/theia-generator/dist/init-sources.js:42:23)
    at Object.next (/home/theia-dev/.yarn-global/node_modules/@eclipse-che/theia-generator/dist/init-sources.js:23:53)
    at /home/theia-dev/.yarn-global/node_modules/@eclipse-che/theia-generator/dist/init-sources.js:17:71
    at new Promise (<anonymous>)
    at __awaiter (/home/theia-dev/.yarn-global/node_modules/@eclipse-che/theia-generator/dist/init-sources.js:13:12)
    at InitSources.readConfigurationAndGenerate (/home/theia-dev/.yarn-global/node_modules/@eclipse-che/theia-generator/dist/init-sources.js:484:16)
    at /home/theia-dev/.yarn-global/node_modules/@eclipse-che/theia-generator/dist/yargs.js:92:54
    at step (/home/theia-dev/.yarn-global/node_modules/@eclipse-che/theia-generator/dist/yargs.js:42:23)
The command '/bin/sh -c che-theia init -c ${HOME}/theia-source-code/che-theia/che-theia-init-sources.yml --alias https://github.com/eclipse/che-theia=${HOME}/theia-source-code/che-theia' returned a non-zero code: 1

**Comment by azatsarynnyy** [comment]
@priyamariarao make sure you are running a correct `build.sh` script as it described in https://github.com/eclipse/che-theia/tree/master/dockerfiles/theia#build-image-manually-using-build-scripts

E.g.:
```
git clone git@github.com:eclipse/che-theia.git && \
cd che-theia && \
./build.sh
```

**Comment by pgirishibm** [comment]
@azatsarynnyy I was working with @priyamariarao  to implement your suggested option.  We got the following working
1) clone the che-theia sources locally
2) build the local che-thiea sources which got us new che-theia images
3) pushed the images to out private image registry
4) Edited the plugin-registry pod to point the default che-theia editor to pick up the custom che-theia image from the private registry
5) created a new workspace. and we got the following error. 
```
Pulling image "us.icr.io/apihub-cr/che-theia:next"
Failed to pull image "us.icr.io/apihub-cr/che-theia:next": rpc error: code = Unknown desc = failed to pull and unpack image "us.icr.io/apihub-cr/che-theia:next": failed to resolve reference "us.icr.io/apihub-cr/che-theia:next": failed to authorize: failed to fetch anonymous token: unexpected status: 401 Unauthorized
Error: Failed to run the workspace: "Unrecoverable event occurred: 'Failed', 'Failed to pull image "us.icr.io/apihub-cr/che-theia:next": rpc error: code = Unknown desc = failed to pull and unpack image "us.icr.io/apihub-cr/che-theia:next": failed to resolve reference "us.icr.io/apihub-cr/che-theia:next": failed to authorize: failed to fetch anonymous token: unexpected status: 401 Unauthorized', 'workspace7oqkni1uh4bfop0i.che-workspace-pod-fd7947f44-srnfr'"
```
The container registry is secured and needs the username/password or secret to access images.   

6) We tried to add the registry secret to the namespace where the workspace pod gets created (which is the same as the che namespace) . But that too did not work. 

7) We then added the container registry credentials in the Administration tab in the che-dashboard. 
<img width="833" alt="image" src="https://user-images.githubusercontent.com/17478559/101612290-7d4d6f00-3a30-11eb-89d4-bddd24aa9967.png">

With this change, the custom che-theia image was pulled and the workspace was started fine. 
We could also see the extension that we had added to the image being loaded and menus getting updated..   

Now the concern with the above change is, it has to be made for every user who logs in. 
Also, if the user get to the dashboard, the user might be able to see the registry credentials and could possibly change it.. 
Another point is, any change to the registry credentials would need to be propagated everywhere. 

We would need a common place where this change will affect all users.   Considering all this, what would be the right approach to add the registry credentials?
